Packet++/src/StpLayer.cpp

```diff
@@ -28,23 +37,23 @@ StpLayer *StpLayer::parseStpLayer(uint8_t *data, size_t dataLen, Layer *prevLaye
 		{
 		case 0x00:
 			return StpConfigurationBPDULayer::isDataValid(data, dataLen)
-						? new StpConfigurationBPDULayer(data, dataLen, prevLayer, packet)
-						: nullptr;
-		case 0x80:
-			return StpTopologyChangeBPDULayer::isDataValid(data, dataLen)
-						? new StpTopologyChangeBPDULayer(data, dataLen, prevLayer, packet)
+						? new StpConfigurationBPDULayer(data, sizeof(stp_conf_bpdu), prevLayer, packet)
```

Comment:
I noticed that while working with crafting functions. Actually, TCN, Configuration BPDU and Rapid STP packets have fixed lengths less than minimum ethernet packet length (only exception MSTP). Every packet is padded with data to match with length requirement, but this padding does not belong to STP layer itself. So i changed this lines to match with only layers itself not with this padding and parse this padding as payload layer

Reply:
Even if that's the case, we should pass `dataLen` as is because it may also affect the parsing of the successive layers. `dataLen` is passed from one layer to another so we don't want to change its value beyond reducing the length of the existing layer.

The length of a certain layer data is determined by `getHeaderLen()`.